gennav/envs/binaryOccupancyGrid2D_env.py

```diff
@@ -0,0 +1,199 @@
+            bool: True if state is valid otherwise False
+        """
+        state = self.transform("world", "map", state)
+        x, y = state.position.x, state.position.y
+        if self.grid[x][-y - 1] == 1:
+            return False
+        else:
+            return True
+
+    def get_traj_status(self, traj):
+        """ Get whether a given trajectory is valid within the environment.
+
+        Method for checking the validity of a trajectory in the given environment.
+
+        Args:
+            state (gennav.utils.Trajectory): Trajectory to be checked
+
+        Returns:
+            bool: True if state is valid otherwise False
+        """
+        collision = False
+        for i in range(len(traj.path) - 1):
+            collision = self.check_line_segment(
+                self.transform("world", "map", traj.path[i]),
+                self.transform("world", "map", traj.path[i + 1]),
+            )
+            if collision:
+                break
+        return not collision
+
+    def transform(self, frame1, frame2, rsf1):
+        """Transform robotPose from one pose to the other
+
+            Args:
+                frame1 (string) : from the frame (world, bot, map)
+                frame2 (string) : to the frame (world, bot, map)
+                rsf1 (gennav.utils.common.RobotState) : RobotState in frame1
+            Returns:
+                rsf2 (gennav.utils.common.RobotState) : RobotState in frame2
+        """
+        # TODO: Make it more robust in terms of checking frames
+
+        # Check if the required trnasform or the inverse of the transform exists
+        frame = frame2 + "T" + frame1
+        frame_inv = frame1 + "T" + frame2
+        if frame in self.transforms.keys():
+            t_matrix = self.transforms[frame]["transform"]
+        elif frame_inv in self.transforms.keys():
+            t_matrix = np.linalg.inv(self.transforms[frame_inv]["transform"])
+        else:
+            raise Exception("Transform for the frames not found")
+
+        # Transform using matrix multiplication
+        pf2 = np.dot(
+            t_matrix, np.array([rsf1.position.x, rsf1.position.y, 1]).reshape(3, 1)
+        )
+        rsf2 = RobotState(position=Point(pf2[0].item(), pf2[1].item()))
+
+        # Return RobotState
+        return rsf2
+
+    def compute_transforms(self):
+        """Computes transforms between frames
+
+            Uses robot pose to compute transform between the world frame and the bot frame
+        """
+        x, y, yaw = (
+            self.robotPose.position.x,
+            self.robotPose.position.y,
+            self.robotPose.orientation.yaw,
+        )
+        worldTbot = np.array(
+            [[np.cos(yaw), -np.sin(yaw), x], [np.sin(yaw), np.cos(yaw), y], [0, 0, 1]]
+        ).reshape(3, 3)
+        self.botTworld["transform"] = np.linalg.inv(worldTbot)
+        self.mapTworld["transform"] = np.dot(
+            self.mapTbot["transform"], self.botTworld["transform"]
+        )
```

Comment:
Are these transforms generalisable? Then we could keep them iin `utils/transforms.py`

Reply:
I dont know if this is the most optimal way. A better way would be to store the transforms properly in a dictionary and recursively search from one frame to another rather than storing each transform separately. For example having botTmap and mapTworld should suffice to obtain transform for botTworld. But here i am storing that separately as a transform, which might become complicated when storing different frames. Although work can be done to store transforms.